File: dbflow/src/main/java/com/raizlabs/android/dbflow/structure/database/OpenHelper.java
package com.raizlabs.android.dbflow.structure.database;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Description: Abstracts out the {@link DatabaseHelperDelegate} into the one used in this library.
 */
public interface OpenHelper {

    void performRestoreFromBackup();

    @NonNull
    DatabaseWrapper getDatabase();

    @Nullable
    DatabaseHelperDelegate getDelegate();

    boolean isDatabaseIntegrityOk();

    void backupDB();

    void setDatabaseListener(@Nullable DatabaseHelperListener helperListener);

    void closeDB();
}

File: dbflow/src/main/java/com/raizlabs/dbflow5/database/OpenHelper.kt
package com.raizlabs.dbflow5.database

import android.os.Build
import android.support.annotation.RequiresApi

/**
 * Description: Abstracts out the [DatabaseHelperDelegate] into the one used in this library.
 */
interface OpenHelper {

    val database: DatabaseWrapper

    val delegate: DatabaseHelperDelegate?

    val isDatabaseIntegrityOk: Boolean

    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
    fun setWriteAheadLoggingEnabled(enabled: Boolean)

    fun performRestoreFromBackup()

    fun backupDB()

    fun setDatabaseListener(helperListener: DatabaseHelperListener?)

    fun closeDB()
}
